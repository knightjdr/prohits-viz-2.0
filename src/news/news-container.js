import PropTypes from 'prop-types';
import React from 'react';
import { Route } from 'react-router-dom';

import Navbar from '../navbar/navbar-container';
import NewsList from './news-list-container';
import NewsItem from './news-item-container';

import './news.css';

const links = [
  {
    route: '/news',
    text: 'NEWS',
  },
];

const NewsContainer = ({
  match,
}) => (
  <div>
    <Navbar links={links} />
    <div className="News-container">
      <Route
        exact
        path={match.path}
        component={NewsList}
      />
      <Route
        path={`${match.path}/:newsId`}
        component={NewsItem}
      />
    </div>
  </div>
);

NewsContainer.propTypes = {
  match: PropTypes.shape({
    path: PropTypes.string,
  }).isRequired,
};

export default NewsContainer;
