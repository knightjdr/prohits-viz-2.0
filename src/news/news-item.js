import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faExclamationTriangle from '@fortawesome/fontawesome-pro-solid/faExclamationTriangle';
import faNewspaper from '@fortawesome/fontawesome-pro-regular/faNewspaper';
import PropTypes from 'prop-types';
import React from 'react';
import { Affix, Spin, Tooltip } from 'antd';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

import NewsItemSelector from '../state/selectors/news-item-selector';
import TextToHtml from '../helpers/text-to-html';

import './news-item.css';

export const NewsItemComponent = ({
  newsItem,
}) => {
  const newsItemElement = newsItem ?
    (
      <div className="News-item-container">
        {
          newsItem.isLoading &&
          <Spin
            size="large"
          />
        }
        {
          newsItem.error &&
          <div>
            <FontAwesomeIcon icon={faExclamationTriangle} />&nbsp;
            There was an error retrieving this story
          </div>
        }
        {
          newsItem.isLoaded &&
          <div className="News-item-content">
            <div className="News-item-navbutton-container">
              <Affix
                className="News-item-navbuttons"
                offsetTop={20}
              >
                <Tooltip
                  placement="right"
                  title="News archive"
                >
                  <NavLink
                    className="News-item-navlink"
                    to="/news"
                  >
                    <FontAwesomeIcon
                      icon={faNewspaper}
                      size="lg"
                    />
                  </NavLink>
                </Tooltip>
              </Affix>
            </div>
            <div className="News-item-story">
              <div className="News-item-headline">
                { newsItem.item.headline }
              </div>
              <div className="News-item-date">
                { newsItem.item.date }
              </div>
              <div className="News-item-details">
                { TextToHtml(newsItem.item.details) }
              </div>
            </div>
          </div>
        }
      </div>
    )
    :
    null;
  return newsItemElement;
};

NewsItemComponent.defaultProps = {
  newsItem: null,
};

NewsItemComponent.propTypes = {
  newsItem: PropTypes.shape({
    error: PropTypes.bool,
    isLoaded: PropTypes.bool,
    isLoading: PropTypes.bool,
    item: PropTypes.shape({
      date: PropTypes.string,
      details: PropTypes.string,
      headline: PropTypes.string,
    }),
  }),
};

const mapStateToProps = state => ({
  newsItem: NewsItemSelector(state),
});

const ConnectedContainer = connect(
  mapStateToProps,
)(NewsItemComponent);

export default ConnectedContainer;
