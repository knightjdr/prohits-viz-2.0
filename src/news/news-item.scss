@import 'theme/_theme.scss';

.News-item-container {
  max-width: 1000px;
}
.News-item-child {
  display: flex;
  flex-direction: column;
  max-width: 950px;
  position: relative;
  top: 0;
}
.News-item-date {
  font-size: 1.2em;
}
.News-item-details {
  margin-top: 20px;
}
.News-item-headline {
  font-size: 1.5em;
  font-weight: bold;
}
.News-item-navbuttons {
  display: flex;
  flex-direction: column;
  left: -40px;
  position: absolute;
  top: 8px;
}
.News-item-navlink,
.News-item-navlink:hover,
.News-item-navlink:visited, {
  color: $accent-dark;
}
