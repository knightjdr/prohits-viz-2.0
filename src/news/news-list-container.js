import React from 'react';

const NewsListContainer = () => (
  <div />
);

export default NewsListContainer;
