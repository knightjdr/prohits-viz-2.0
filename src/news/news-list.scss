@import 'theme/_mixins.scss';

.News-list-container {
  max-width: 1000px;
  width: calc(100vh - 40px);
}
.News-list-container .ant-list-item-meta-title {
  font-family: TitleFont;
  text-transform: uppercase;
}
.News-list-child {
  display: flex;
  flex-direction: column;
}
.News-list-details {
  @include line-clamp(2);
  text-align: justify;
}
.News-list-pagination {
  float: right;
}
.News-list-title {
  font-family: TitleFont;
  font-size: 2em;
  font-weight: bold;
}
