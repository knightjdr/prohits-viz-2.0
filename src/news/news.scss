.News-container {
  display: flex;
  justify-content: center;
  max-width: 1000px;
  padding: 10px 10px;
}
