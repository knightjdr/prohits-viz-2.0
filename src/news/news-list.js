import React from 'react';

import './news.css';

const News = () => (
  <div className="News-container" />
);

export default News;
