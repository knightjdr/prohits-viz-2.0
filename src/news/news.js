import PropTypes from 'prop-types';
import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';

import Navbar from '../navbar/navbar-container';
import NewsList from './news-list/news-list-container';
import NewsItem from './news-item/news-item-container';
import RouteNotFound from '../router/route-not-found';

import './news.css';

const links = [
  {
    route: '/news',
    text: 'NEWS',
  },
];

export const NewsComponent = ({
  match,
}) => (
  <div>
    <Navbar links={links} />
    <div className="News-container">
      <Switch>
        <Route
          exact
          path={match.path}
          component={NewsList}
        />
        <Route
          exact
          path={`${match.path}/:newsId`}
          component={NewsItem}
        />
        <RouteNotFound />
      </Switch>
    </div>
  </div>
);

NewsComponent.propTypes = {
  match: PropTypes.shape({
    path: PropTypes.string,
  }).isRequired,
};

export default withRouter(NewsComponent);
