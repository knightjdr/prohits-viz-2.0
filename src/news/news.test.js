import React from 'react';
import { shallow } from 'enzyme';

import { NewsComponent } from './news';

const route = {
  noId: {
    path: '/news',
  },
  withId: {
    path: '/news/id',
  },
};

describe('News', () => {
  test('Renders list when no ID specified', () => {
    const wrapper = shallow(
      <NewsComponent
        match={route.noId}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Renders item when ID specified', () => {
    const wrapper = shallow(
      <NewsComponent
        match={route.withId}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
