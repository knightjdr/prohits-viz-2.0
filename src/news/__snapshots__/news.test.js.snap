// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Renders item when ID specified 1`] = `
<div>
  <NavbarContainer
    background={true}
    links={
      Array [
        Object {
          "route": "/news",
          "text": "NEWS",
        },
      ]
    }
  />
  <div
    className="News-container"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/news/id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/news/id/:newsId"
      />
      <RouteNotFound />
    </Switch>
  </div>
</div>
`;

exports[`News Renders list when no ID specified 1`] = `
<div>
  <NavbarContainer
    background={true}
    links={
      Array [
        Object {
          "route": "/news",
          "text": "NEWS",
        },
      ]
    }
  />
  <div
    className="News-container"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/news"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/news/:newsId"
      />
      <RouteNotFound />
    </Switch>
  </div>
</div>
`;
